app/Http/Controllers/ActionStatsController.php

```diff
@@ -32,12 +32,29 @@ public function index(Request $request)
         $query = $this->newQuery(ActionStat::class);
 
         $filters = $request->query('filter');
-        $query = $this->filter($query, $filters, ActionStat::$indexes);
+
+        /**
+         * Because we may be joining on groups, which also have location and school_id columns, we
+         * specify our table name to avoid integrity constraint violations for ambiguous clauses.
+         */
+        $tableName = $query->getModel()->getTable();
+
+        if (Arr::has($filters, 'action_id')) {
+            $query->where('action_id', $filters['action_id']);
+        }
+
+        if (Arr::has($filters, 'location')) {
+            $query->where($tableName . '.location', $filters['location']);
+        }
```

Comment:
Instead of hard-coding this solution here for each filter, can we add this table name prefix to the `filter` method itself? We'll always want to be filtering on the "primary" model I think (and that'd make sure we don't have to continue to make one-off change as other cases like this occur).

Reply:
Hmm, the `filter` method actually hands off to [this helper](https://github.com/DoSomething/rogue/blob/989450c162efe4cf444c7acd5ccce98a0e091355/app/helpers.php#L60-L64) so I guess the fix would go there (although TBH I think we could just inline this in the two places it's referenced to make things a little easier to follow).

Reply:
Ah forgot that [`filter` method is custom.](https://github.com/DoSomething/rogue/blob/989450c162efe4cf444c7acd5ccce98a0e091355/app/Http/Controllers/Traits/FiltersRequests.php#L27). I'll merge and deploy this for now to stop the noise (and also see if this super mysterious  error goes away) and follow up in a new PR.